features/step_definitions/supplier_utils.rb

```diff
@@ -0,0 +1,138 @@
+  supplier_data["suppliers"]["contactInformation"][0]["postcode"] = supplier_postcode
+  supplier_data["suppliers"]["dunsNumber"] = supplier_dunsnumber
+
+  supplier_path = "/suppliers/#{supplier_id}"
+
+  response = call_api(:get, supplier_path)
+  if response.code == 404
+    response = call_api(:put, supplier_path, payload: supplier_data)
+    response.code.should be(201), response.body
+  else
+    response = call_api(:post, supplier_path, payload: supplier_data)
+    response.code.should be(200), response.body
+
+    contact_id = JSON.parse(response.body)["suppliers"]["contactInformation"][0]["id"]
+    contact_data = JSON.parse("{\"updated_by\":\"Functional tests\", \"contactInformation\":#{supplier_data ["suppliers"]["contactInformation"][0].to_json}}")
+    response = call_api(:post, "/suppliers/#{supplier_id}/contact-information/#{contact_id}", payload: contact_data, safe_for_smoke_tests: true)
+    response.code.should be(200), response.body
+  end
+end
+
+# services
+def create_live_service (supplier_id, service_id, framework, lot)
+  file = File.read("./fixtures/#{framework}-#{lot}-service.json")
+  service_data = JSON.parse(file)
+  service_data["services"]["id"] = service_id
+  service_data["services"]["supplierId"] = supplier_id
+  service_data["services"]["serviceName"] = "#{service_id} #{service_data["services"]["serviceName"]}" if framework.start_with? "g-cloud"
+
+  service_path = "/services/#{service_id}"
+
+  response = call_api(:get, service_path)
+  if response.code == 404
+    response = call_api(:put, service_path, payload: service_data)
+    response.code.should be(201), response.body
+  else
+    response = call_api(:post, service_path, payload: service_data)
+    response.code.should be(200), response.body
+
+    service_data.delete("services")
+    response = call_api(:post, service_path + "/status/published", payload: service_data)
+    response.code.should be(200), response.body
+  end
+end
+
+def service_status_public (service_id)
+  page.visit("#{dm_frontend_domain}/admin/services/#{service_id}")
+  page.should have_content('Service status')
+  current_service_status = ''
+  while current_service_status != 'Public'
+    current_service_status = page.find(:xpath,"//*[contains(text(), 'Service status')]/following-sibling::*[contains(@class, 'selection-button') and contains(@class ,'selected')][text()]").text()
+    if current_service_status == 'Removed'
+      update_and_check_status("Private")
+    elsif current_service_status == 'Private'
+      update_and_check_status("Public")
+    end
+  end
+end
+
+def login_page_type(user_type)
+  case user_type
+  when "CCS Sourcing", "CCS Category", "Administrator"
+    return "Administrator"
+  when "Supplier", "Buyer"
+    return "Digital Marketplace"
+  else
+    fail("User type \"#{user_type}\" does not exist")
+  end
+end
+
+Given(/I have logged in to Digital Marketplace as a '(.*)' user$/) do |user_type|
```

Comment:
We have [login steps already](https://github.com/alphagov/digitalmarketplace-functional-tests/blob/master/features/step_definitions/login_steps.rb), which you should try and use instead of creating new ones.